app/Http/Controllers/Api/v1/EntityRecoveryApiController.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Http\Controllers\Api\v1;
+
+use App\Models\Campaign;
+use App\Models\Entity;
+use App\Http\Resources\EntityResource as Resource;
+use App\Http\Requests\RecoverEntity as Request;
+use App\Services\RecoveryService;
+
+class EntityRecoveryApiController extends ApiController
+{
+    /** @var RecoveryService */
+    protected $service;
```

Comment:
You can type the service and remove the comment :-)
`protected RecoveryService $service;`